static/js/app.js

```diff
@@ -401,6 +401,46 @@ window.onerror = function reportClientException(message, source, line_number, co
   });
 }
 
+function convert_lang_to_ISO(lang){
+switch (lang){
+    case 'nl':
+        return 'nl-NL';
+    case 'hu':
+        return 'hu-HU';
+    case 'sw':
+        return 'sw-KE';
+    case 'fr':
+        return 'fr-FR';
+    case 'pt':
+        return 'pt-BR';
+    case 'de':
+        return 'de-DE';
+    case 'it':
+        return 'it-IT';
+    case 'el':
+        return 'el-GR';
+    case 'zh':
+        return 'zh-CN';
+    case 'es':
+        return 'es-ES';
+    default:
+        return 'en-US';
+
+}
+}
+
+function speak(text) {
+    speak_enabled = $('#speak').is(":checked")
+    console.log('speak?', speak_enabled);
+    if (speak_enabled){
+        let utterance = new SpeechSynthesisUtterance(text);
+        lang = window.State.lang;
+        utterance.lang = convert_lang_to_ISO(lang);
```

Comment:
Some testing finds that this doesn't work on Safari, unfortunately.

We're supposed to use the `utterance.voice = ` field there, apparently.

How about:

```ts
function findVoice(langWithCountry) {
  // If the feature doesn't exist in the browser, return null
  if (!window.speechSynthesis) { return undefined; }
  const ourVoices = window.speechSynthesis.getVoices().filter(voice => voice.lang === langWithCountry);
  
  // Give preference to local voices to not send Chrome users a big data bill (otherwise
  // Google voices will be the first in the list)
  const localVoices = ourVoices.filter(voice => voice.localService);
  if (localVoices.length > 0) { return localVoices[0]; }

  // Otherwise any voice we can get our hands on
  return ourVoices.length > 0 ? ourVoices[0] : undefined;
}
```